retrofit-adapters/rxjava/src/main/java/retrofit/RxJavaCallAdapterFactory.java

```diff
@@ -173,13 +182,8 @@ private CallOnSubscribe(Call<T> originalCall) {
       return responseType;
     }
 
-    @Override public <R> Observable<Result<R>> adapt(Call<R> call) {
-      return Observable.create(new CallOnSubscribe<>(call)) //
-          .map(new Func1<Response<R>, Result<R>>() {
-            @Override public Result<R> call(Response<R> response) {
-              return Result.response(response);
-            }
-          })
+    @Override public <R> Observable<Result<R>> adapt(Call<R> call, Retrofit retrofit) {
+      return Observable.create(new CallOnSubscribe<>(call, retrofit)) //
           .onErrorReturn(new Func1<Throwable, Result<R>>() {
```

Comment:
Oh, no it can't. But it _could_. I find it a bit strange that `CallOnSubscribe` emits `Result` instances for responses but not for errors. I think we probably want it to conform to always emitting a `Result` instance and then have these functions decompose its instances.